server/genserver.go

```diff
@@ -205,15 +205,15 @@ func New(c Controller, addr string) *Server {
 	return NewWithMiddleware(c, addr, []func(http.Handler) http.Handler{})
 }
 
-// NewWithMiddleware returns a Server that implemenets the Controller interface. It runs the
-// middleware after the built-in middleware (e.g. logging), but before the controller methods.
-// The middleware is executed in the order specified. The server will start when "Serve" is called.
-func NewWithMiddleware(c Controller, addr string, m []func(http.Handler) http.Handler) *Server {
+// NewRouter returns a Handler (more specifically mux.Router) with no middleware
+func NewRouter(c Controller) http.Handler {
```

Comment:
I would change this to return the `*mux.Router` to make it clear that we're exposing it to attach additional routes.

I'd also mention that in the comment as well